src/components/DeliverableView/InternalComment.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { Grid, Typography } from '@mui/material';
+import TextField from '@terraware/web-components/components/Textfield/Textfield';
+import strings from 'src/strings';
+import theme from 'src/theme';
+import { Deliverable } from 'src/types/Deliverables';
+import useSnackbar from 'src/utils/useSnackbar';
+import Button from 'src/components/common/button/Button';
+import DialogBox from 'src/components/common/DialogBox/DialogBox';
+import useUpdateDeliverable from 'src/scenes/AcceleratorRouter/useUpdateDeliverable';
+
+interface InternalCommentProps {
+  deliverable: Deliverable;
+}
+
+const InternalComment = ({ deliverable }: InternalCommentProps) => {
+  const snackbar = useSnackbar();
+  const { status, update } = useUpdateDeliverable();
+
+  const [isDialogOpen, setIsDialogOpen] = useState<boolean>(false);
+  const [internalComment, setInternalComment] = useState(deliverable.internalComment || '');
+
+  const toggleDialog = useCallback(() => {
+    setIsDialogOpen((prev) => !prev);
+  }, []);
+
+  const handleUpdate = () => {
+    update({ ...deliverable, internalComment });
```

Comment:
Forgoing updating the types (for the `UpdateRequest`) through the update flow until we get the types from the backend.